### PR TITLE
Upgrade version update

### LINKED ISSUE #3420: Update versions between migrations
## Problem Definition

Given
https://github.com/raiden-network/raiden/blob/d50045c7aec544d225dcddc4fb2ee81f35e1949e/raiden/utils/upgrades.py#L125-L135

And assuming that each migration functions upgrades from one version to another **completely***, the current implementation does not update the version after each iteration which causes a problem when properly checking whether a migration is upgrading from the correct "old" version to the correct "new" version.

### Proposed Solution
The upgrade manager should iterate versions from `older_version` all the way up to `RAIDEN_DB_VERSION` and after executing an upgrade function, the iteration version number should be increased until `RAIDEN_DB_VERSION` is reached or there are no more upgrade functions to be executed.

Related issue #3405

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #3420

### CODE REVIEW

**Comment by LefterisJP** [comment]
Reviewing this.

File: raiden/tests/unit/test_upgrade.py [hunk 1]
```diff
@@ -48,3 +50,41 @@ def test_upgrade_manager_restores_backup(tmp_path):
     assert state_change_record.data is not None
     assert not old_db_filename.exists()
     assert Path(str(old_db_filename).replace('_log.db', '_log.backup')).exists()
+
+
+def test_sequential_version_numbers(tmp_path):
```
Comment (LefterisJP) [inline raiden/tests/unit/test_upgrade.py]: nitpick: Maybe just add docstring here explaining what the test does? Thanks to our call today I know exactly what you want to test but perhaps explain for the others too?

Something like:
```
"""
Test that when there are multiple DB upgrades to be applied each one is ran and only once.
"""
```

File: raiden/utils/upgrades.py [hunk 2]
```diff
@@ -120,21 +121,32 @@ def run(self):
...
 
-                    update_version(cursor)
+                    update_version(cursor, RAIDEN_DB_VERSION)
                     # Prevent the upgrade from happening on next restart
                     self._backup_old_db(old_db_filename)
                 except RaidenDBUpgradeError:
                     self._delete_current_db()
                     raise
 
+    def _run_upgrade_func(self, cursor: sqlite3.Cursor, func: Callable, version: int):
```
Comment (LefterisJP) [inline raiden/utils/upgrades.py]: add return type: `: -> int`

File: raiden/tests/unit/test_upgrade.py (hunk 1 above)
Comment (rakanalh) [reply raiden/tests/unit/test_upgrade.py]: done

**Pull request merged**